# Marketplace Customer Analytics

## Regional Analysis

This notebook visualizes regional customer and order statistics
obtained from the SQL analysis of marketplace data.

The analysis compares:
- customer base size;
- total number of orders;
- average order value (AOV);
- installment usage;
- promo code usage;
- customer cancellation rate.

The underlying SQL query is available in:

`../sql/adhoc03_regional_statistics.sql`

In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("regional_statistics.csv")

df

,region,total_users_region,orders_total_region,aov_region,installments_ratio,promo_ratio,cancel_ratio
0,Москва,39386,40747,3140.14,0.48,0.037,0.0063
1,Санкт-Петербург,11978,12414,3593.46,0.55,0.042,0.0053
2,Новосибирская область,11044,11401,3491.79,0.54,0.037,0.0043


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   region               3 non-null      str    
 1   total_users_region   3 non-null      int64  
 2   orders_total_region  3 non-null      int64  
 3   aov_region           3 non-null      float64
 4   installments_ratio   3 non-null      float64
 5   promo_ratio          3 non-null      float64
 6   cancel_ratio         3 non-null      float64
dtypes: float64(4), int64(2), str(1)
memory usage: 300.0 bytes


In [6]:
df.isna().sum()

region                 0
total_users_region     0
orders_total_region    0
aov_region             0
installments_ratio     0
promo_ratio            0
cancel_ratio           0
dtype: int64

In [7]:
fig = px.bar(
    df.sort_values("total_users_region", ascending=True),
    x="total_users_region",
    y="region",
    orientation="h",
    title="Customer Base by Region",
    labels={
        "total_users_region": "Number of Customers",
        "region": "Region"
    },
    text="total_users_region"
)

fig.update_traces(textposition="outside")

fig.show()

In [26]:
fig = px.bar(
    df.sort_values("aov_region", ascending=True),
    x="aov_region",
    y="region",
    orientation="h",
    title="Average Order Value (AOV) by Region",
    labels={
        "aov_region": "Average Order Value",
        "region": "Region"
    },
    text="aov_region"
)

fig.update_traces(texttemplate="%{text:.2f}", textposition="outside")

fig.show()

In [22]:
fig = px.bar(
    df.sort_values("installments_ratio", ascending=True),
    x="installments_ratio",
    y="region",
    orientation="h",
    title="Installment Usage by Region",
    labels={
        "installments_ratio": "Share of Orders with Installments",
        "region": "Region"
    },
    text="installments_ratio"
)

fig.update_traces(
    texttemplate="%{text:.2%}",
    textposition="inside"
)

fig.show()

In [27]:
fig = px.bar(
    df.sort_values("promo_ratio", ascending=True),
    x="promo_ratio",
    y="region",
    orientation="h",
    title="Promo Code Usage by Region",
    labels={
        "promo_ratio": "Share of Orders with Promo Code",
        "region": "Region"
    },
    text="promo_ratio"
)

fig.update_traces(
    texttemplate="%{text:.2%}",
    textposition="inside"
)

fig.show()

In [24]:
fig = px.bar(
    df.sort_values("cancel_ratio", ascending=True),
    x="cancel_ratio",
    y="region",
    orientation="h",
    title="Customer Cancellation Rate by Region",
    labels={
        "cancel_ratio": "Share of Customers with Cancelled Orders",
        "region": "Region"
    },
    text="cancel_ratio"
)

fig.update_traces(
    texttemplate="%{text:.2%}",
    textposition="inside"
)

fig.show()

In [25]:
df.sort_values("total_users_region", ascending=False)

,region,total_users_region,orders_total_region,aov_region,installments_ratio,promo_ratio,cancel_ratio
0,Москва,39386,40747,3140.14,0.48,0.037,0.0063
1,Санкт-Петербург,11978,12414,3593.46,0.55,0.042,0.0053
2,Новосибирская область,11044,11401,3491.79,0.54,0.037,0.0043


## Key Findings

- Moscow has the largest customer base and the highest total number of orders among the analyzed regions.
- Saint Petersburg has the highest average order value (AOV) at 3,593.46 and the highest share of orders paid in installments at 55%.
- Novosibirsk region has a similar AOV to Saint Petersburg, while Moscow has a lower average order value despite having the largest customer base.
- Promo code usage is relatively similar across regions, ranging from 3.7% to 4.2%.
- Moscow has the highest customer cancellation rate at 0.63%, although cancellation rates remain low across all analyzed regions.